src/runtime/static.ts

```diff
@@ -15,38 +15,41 @@ export default eventHandler((event) => {
     return;
   }
 
-  let id = decodeURIComponent(
-    withLeadingSlash(
-      withoutTrailingSlash(parseURL(event.node.req.url).pathname)
-    )
-  );
+  let id: string;
   let asset;
+  try {
+    id = decodeURIComponent(
+      withLeadingSlash(
+        withoutTrailingSlash(parseURL(event.node.req.url).pathname)
+      )
+    );
 
-  const encodingHeader = String(
-    event.node.req.headers["accept-encoding"] || ""
-  );
-  const encodings = [
-    ...encodingHeader
-      .split(",")
-      .map((e) => EncodingMap[e.trim()])
-      .filter(Boolean)
-      .sort(),
-    "",
-  ];
-  if (encodings.length > 1) {
-    event.node.res.setHeader("Vary", "Accept-Encoding");
-  }
+    const encodingHeader = String(
+      event.node.req.headers["accept-encoding"] || ""
+    );
+    const encodings = [
+      ...encodingHeader
+        .split(",")
+        .map((e) => EncodingMap[e.trim()])
+        .filter(Boolean)
+        .sort(),
+      "",
+    ];
+    if (encodings.length > 1) {
+      event.node.res.setHeader("Vary", "Accept-Encoding");
+    }
 
-  for (const encoding of encodings) {
-    for (const _id of [id + encoding, joinURL(id, "index.html" + encoding)]) {
-      const _asset = getAsset(_id);
-      if (_asset) {
-        asset = _asset;
-        id = _id;
-        break;
+    for (const encoding of encodings) {
+      for (const _id of [id + encoding, joinURL(id, "index.html" + encoding)]) {
+        const _asset = getAsset(_id);
+        if (_asset) {
+          asset = _asset;
+          id = _id;
+          break;
+        }
       }
     }
-  }
+  } catch {}
```

Comment:
Why we need a try catch here with wide scope?

Reply:
It was so we could avoid further normalising of the id if it couldn't be resolved. I'll push a refactor with `decodePath`.

Reply:
Thanks. Would be nice if we could limit the scope of try/catch this way to avoid hiding other errors.